main.c

```diff
@@ -237,7 +241,66 @@ int NMEA_message_handler(int sock)
 }
 
 /**
-* @brief Timming routine for pressure measurement
+* @brief Timing routine for temperature measurement
+* @param 
+* @return 
+* 
+* Timing handler to coordinate temperature measurement
+* @date 17.04.2014 born
+*
+*/ 
+int temperature_measurement_handler(int sock)
+{
+
+	static int temp_counter = 0,done;
+	char s[70];
+	int sock_err = 0;
+	int result;
+
+	if (temp_sensor.present) {
+		if (temp_counter==0) {
+			if (OWReset(&temp_sensor)==1) { // Reset
+				if (OWWriteByte(&temp_sensor,0xCC)==1) { // Skip ROM
+					if (OWWriteByte(&temp_sensor,0x44)==1) result=1; // Initiate Conversion
+				}
+			}
+			done=0;
+			temp_counter=1;
+		} else {
+			if (!done) {
+				if (OWReadByte(&temp_sensor)>0) {
+					if (OWReset(&temp_sensor)==1) {  // Reset
+						if (OWWriteByte(&temp_sensor,0xCC)==1) { // Skip ROM
+							if (OWWriteByte(&temp_sensor,0xBE)==1) { //Read Scratchpad
+								OWReadTemperature(&temp_sensor); // Convert output to temperature
+								if (temp_sensor.valid==1) {
+									done=1;
+									// Compose POV NMEA sentences
+									result=Compose_Temperature_POV(&s[0], temp_sensor.temperature);
```

Comment:
Please do not compose the NMEA sentence in the measurement handler routine.
Just sample the value, but compose the NMEA sentence in nmea_handler.